libcalico-go/lib/epstatusfile/status_file_watcher.go

```diff
@@ -167,6 +164,9 @@ func (w *FileWatcher) runWatcher() {
 			log.WithError(err).Info("Error initializing fsnotify. Falling back to polling.")
 		}
 
+		// Get current state of the directory and emit initial events.
+		w.scanDirectory()
```

Comment:
This ends up scanning the directory twice per loop if we're polling. I think we need a scan _here_ to make sure that we don't miss any events so perhaps just remove the one in the 

```
case <-w.pollTicker.C:
``` 
arm of the select below?  Then the poll ticker will just continue the loop, which will try to make a new watcher and then do this poll.

Reply:
Sorted. I also noticed we're not closing the watcher anywhere when we re-iterate, so I added that code too under `newFsnotifyWatcher()`